### DISCUSSION TITLE
Port forwarding: Issues when transferring many ressources

### DISCUSSION THREAD

**Discussion by max06** [community/community#57643] [Codespaces]
### Select Topic Area

Bug

### Body

While trying to load the vscode web-ui inside a codespace (the dev-version, for contributing), I was almost always rewarded with a white page. The js-console showed many failed ressources labeled with `net::ERR_HTTP2_SERVER_REFUSED_STREAM`.

I got it working when I activated the network throttling (set to `Fast 3G`). 

(I'm not talking about the vscode-UI connected to the codespace!)

Reproduction:
Run a codespace (big instance, 4C/8GB) with the microsoft/vscode repository.
Run `yarn && yarn compile`, then start the debug task for VSCode Web.
Open the link with the token in the debug output panel.
You can see around 1180 Ressources being loaded in the devtools, but the app isn't starting. You'll find errors in the js console.

**Comment by joshspicer** [comment]
Hi @max06, thanks for the feedback!  You're saying that throttling your network speed mitigates what you're seeing?

I'm not too familiar with development on the vscode repo - are you saying after a `yarn compile` a server to connect to a dev version of vscode-web is started automatically on a port, and connecting to that port doesn't work (unless you throttle your network speed?).

Also, can you let me know if you're connecting to your codespace from VS Code desktop or via your web browser?

**Reply by max06** [reply]
Hello @joshspicer 
quick reaction! 
The `yarn compile` is required before running the app - the debug task "VSCode Web" actually launches the server on the preconfigured port. The server hosts the web UI, besides some other vscode internal stuff like the extension host. 
The forwarded port works, it just doesn't seem to be able to keep up with the huge amount of requests being initiated by the UIs main page. It works better/produces less errors if I throttle my connection in my devtools.

I'm primarily using codespaces from my browser - it's one step less from github.

If you're interested, I can give you a live demo - I'd drop you an invitation on twitter in that case.

**Comment by benny-png** [comment]
Just use port-forwarding services like NGROK lol SUPER easy three line of shell commands https://ngrok.com/download
you can use this for reference https://youtu.be/AhiXVaWst24

**Reply by max06** [reply]
![image](https://github.com/community/community/assets/7556827/57959831-2a45-49f6-bdc1-1c1c3343a059)

**Comment by AvivonCtrl** [comment]
Did you end up finding a solution?
Running Vite with chunking results in like 100 requests being made and apparently I keep getting the same `net::ERR_HTTP2_SERVER_REFUSED_STREAM`, likewise setting to Fast 3G in networking seems to fix this.

Btw I'm using VSCode Desktop.

**Reply by max06** [reply]
I switched to my local docker daemon ... only the workaround is helping so far to my knowledge.

**Reply by AvivonCtrl** [reply]
I'm able to re-produce 100% of the times.
Simply make requests > 20 at once, which is a classic chunking // Vite feature.
